pkg/fs/lock.go

```diff
@@ -0,0 +1,66 @@
+package fs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const lockFilename = "LOCK"
+
+// ClaimLock on the specified path.
+// Successful claims must eventually be released.
+func ClaimLock(fs Filesystem, path string) error {
+	lockFile := filepath.Join(path, lockFilename)
+	if fs.Exists(lockFile) {
```

Comment:
Yep, excellent observation. Unfortunately, realFilesystem.Create calls os.Create, which "creates the named file with mode 0666 (before umask), **truncating it if it already exists.**". And I didn't immediately see a non-racy way to get Create-if-not-exists semantics. Perhaps I'll have to upgrade to something like [prometheus/util/flock](https://godoc.org/github.com/prometheus/prometheus/util/flock)...